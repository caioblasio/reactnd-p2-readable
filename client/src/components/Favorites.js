import React, { Fragment } from 'react'
import { withStyles } from '@material-ui/core/styles';
import { Link } from 'react-router-dom';
import PostList from './PostList';
import Button from '@material-ui/core/Button';
import AddIcon from '@material-ui/icons/Add';
import { getValue } from '../utils/locaStorage';
import PropTypes from 'prop-types';

const styles = theme => ({
  button: {
    position: 'fixed',
    bottom: theme.spacing.unit * 3,
    right: theme.spacing.unit * 3,
  }
});

/**
 * @description Favorite component that renders post list marked as favorites
 * @param {object} classes
*/
const Favorites = props => {

  const { classes } = props;

  return (
    <Fragment>
      <PostList includeOnlyIds={getValue('favorites', true)} headline="Favorites" />
      <Link to="/new">
        <Button variant="fab" color="primary" aria-label="Add" className={classes.button}>
          <AddIcon />
        </Button>
      </Link>
    </Fragment>
  )
  
}

Favorites.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(Favorites)