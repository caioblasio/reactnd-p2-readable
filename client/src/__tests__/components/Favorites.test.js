import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Favorites from '../../components/Favorites';


describe('<Favorites />', () => {

    it('renders the component', () => {
      const wrapper = shallow(<Favorites />);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });

});