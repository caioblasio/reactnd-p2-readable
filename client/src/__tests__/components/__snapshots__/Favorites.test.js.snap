// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Favorites /> renders the component 1`] = `
<Fragment>
  <WithStyles(Connect(PostList))
    headline="Favorites"
    includeOnlyIds={Array []}
  />
  <Link
    replace={false}
    to="/new"
  >
    <WithStyles(Button)
      aria-label="Add"
      className="Favorites-button-1"
      color="primary"
      variant="fab"
    >
      <pure(AddIcon) />
    </WithStyles(Button)>
  </Link>
</Fragment>
`;
